templates/master/01-master-kubelet/_base/files/kubelet.yaml

```diff
@@ -26,6 +26,7 @@ contents:
     staticPodPath: /etc/kubernetes/manifests
     systemCgroups: /system.slice
     featureGates:
+      AlibabaPlatform: true
```

Comment:
This is might be a dumb question, but if we're processing feature gates properly, and Joel already added it to the defaults e.g. 
https://github.com/openshift/machine-config-operator/blob/1039029c654795af9af1ef3d3c916075bc147c30/vendor/github.com/openshift/api/config/v1/types_feature.go#L194 which is vendored as part of this,  what is the intent behind also including it here?  

(Is it just "make sure that if we don't have our featuregate ducks in a row that AlibabaPlatform doesn't get shut off anywhere yet" ? )

Reply:
good questions, @rphillips might know the answer of how it is all wired up. I needed to add it as kubelet `TestFeatureGateDrift` test was failing with recent change you have mentioned which was part of latest PR https://github.com/openshift/api/pull/1478/

Reply:
Reading from original PR https://github.com/openshift/machine-config-operator/pull/187 when featureGate field was added, it seems like these fields get used during initial bootstarpping process and help kubelet decide which features to turn on.

Reply:
@jkyros  All featureGates mentioned in kubelet template matches what is defined as defaultFeatures in  https://github.com/openshift/api/blob/master/config/v1/types_feature.go#L191 . Do you see any concern here with adding alibabaPlatform to match with it ?

Reply:
So the issues with these tests, IIUC, is that they have a statically coded list of features, which needs updating every time the default feature set changes. That is going to happen often! It would be better if we can update the tests/templates so that there is no checked in default set of features, else this will need constant updates, but I'm not sure how possible that is

> "make sure that if we don't have our featuregate ducks in a row that AlibabaPlatform doesn't get shut off anywhere yet"

Yes, the intention is to move all code for alibaba behind a feature gate so that we can then switch it off with a single PR and prove that nothing breaks by doing so